lightning/src/util/zbase32.rs

```diff
@@ -0,0 +1,142 @@
+// This is a modification of base32 encoding to support the zbase32 alphabet.
+// The original piece of software can be found at https://github.com/andreasots/base32
+// The original portions of this software are Copyright (c) 2015 The base32 Developers
+
+/* This file is licensed under either of
+ *  Apache License, Version 2.0, (LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0) or
+ *  MIT license (LICENSE-MIT or http://opensource.org/licenses/MIT)
+ * at your option.
+*/
+
+const ALPHABET: &'static [u8] = b"ybndrfg8ejkmcpqxot1uwisza345h769";
+
+/// Encodes some bytes as a zbase32 string
+pub fn encode(data: &[u8]) -> String {
+	let mut ret = Vec::with_capacity((data.len() + 4) / 5 * 8);
+
+	for chunk in data.chunks(5) {
+		let buf = {
+			let mut buf = [0u8; 5];
+			for (i, &b) in chunk.iter().enumerate() {
+				buf[i] = b;
+			}
+			buf
+		};
+
+		ret.push(ALPHABET[((buf[0] & 0xF8) >> 3) as usize]);
+		ret.push(ALPHABET[(((buf[0] & 0x07) << 2) | ((buf[1] & 0xC0) >> 6)) as usize]);
+		ret.push(ALPHABET[((buf[1] & 0x3E) >> 1) as usize]);
+		ret.push(ALPHABET[(((buf[1] & 0x01) << 4) | ((buf[2] & 0xF0) >> 4)) as usize]);
+		ret.push(ALPHABET[(((buf[2] & 0x0F) << 1) | (buf[3] >> 7)) as usize]);
+		ret.push(ALPHABET[((buf[3] & 0x7C) >> 2) as usize]);
+		ret.push(ALPHABET[(((buf[3] & 0x03) << 3) | ((buf[4] & 0xE0) >> 5)) as usize]);
+		ret.push(ALPHABET[(buf[4] & 0x1F) as usize]);
+	}
+
+	ret.truncate((data.len() * 8 + 4) / 5);
+
+	// Check that our capacity calculation doesn't under-shoot in fuzzing
+	#[cfg(fuzzing)]
+	assert_eq!(ret.capacity(), (data.len() + 4) / 5 * 8);
+
+	String::from_utf8(ret).unwrap()
+}
+
+// ASCII 0-Z
+const INV_ALPHABET: [i8; 43] = [
+	-1, 18, -1, 25, 26, 27, 30, 29, 7, 31, -1, -1, -1, -1, -1, -1, -1,  24, 1, 12, 3, 8, 5, 6, 28,
+	21, 9, 10, -1, 11, 2, 16, 13, 14, 4, 22, 17, 19, -1, 20, 15, 0, 23,
+];
+
+/// Decodes a zbase32 string to the original bytes, failing if the string was not encoded by a
+/// proper zbase32 encoder.
+pub fn decode(data: &str) -> Result<Vec<u8>, ()> {
```

Comment:
This work well `pub(super)`, no need to make them public ?

Reply:
I made it public so it can be easily bound with methods that import `lightning` instead of working within the same repo for the bindings, like the Python bindings do. Happy do discuss it though.

Reply:
Does that mean you need zbase32 for other uses, or just generally you think people may need zbase32 for other things and want it via bindings?

Reply:
It may not be needed for anything but signing, so I guess we could make it private.